src/Common/src/System/IO/DriveInfoInternal.Unix.cs

```diff
@@ -10,6 +10,13 @@ namespace System.IO
     /// <summary>Contains internal volume helpers that are shared between many projects.</summary>
     internal static partial class DriveInfoInternal
     {
+#if MONO
+    static class Unix
+    {
+#endif
```

Comment:
No it can't as described in https://github.com/dotnet/corefx/pull/17840#issuecomment-292292346. To be more specific SpecialFolder (from `Environment.SpecialFolder.cs`) is nested into `Environment`, so by renaming the class from `Environment` to `EnvironmentUnix` in `Environment.Unix.cs` we lose access to `SpecialFolder`.